File: solutions/java/reverse-string/2/src/main/java/ReverseString.java
class ReverseString {

    String reverse(String inputString) {
        char[] charStr = inputString.toCharArray();
        int len = charStr.length;
        for(int i=0; i<len/2; i++) {
        	char temp = charStr[i];
        	charStr[i] = charStr[len-i-1];
        	charStr[len-i-1] = temp;
        }
		return String.copyValueOf(charStr);
    }
  
}

File: solutions/cpp/reverse-string/2/reverse_string.cpp
#include <algorithm>
#include <string>
#include "reverse_string.h"
namespace reverse_string 
{
        std::string reverse_string(const char* st){
        
        std::string str{st};
        size_t s = str.size();
        for(size_t i{}; i<s/2; i++){
            char t = str[i];
            str[i] = str[s-i-1];
            str[s-i-1] = t;
            // std::cout<<str[s-i-1]<<std::endl;
        }
        return str;
    }
}